File: 🟢 Easy/2418 - Sort the People/_2418SortThePeople.java
class Solution {

    class People {
        String name;
        int height;

        People(String name, int height) {
            this.name = name;
            this.height = height;
        }
    }

    class HeightComparator implements Comparator<People> {
        public int compare(People p1, People p2) {
            return p2.height - p1.height;
        }
    }

    public String[] sortPeople(String[] names, int[] heights) {
        ArrayList<People> nameHeight = new ArrayList<>();

        for (int i = 0; i < names.length; i++) {
            nameHeight.add(new People(names[i], heights[i]));
        }

        Collections.sort(nameHeight, new HeightComparator());

        int i = 0;
        for (People sortedName : nameHeight) {
            names[i++] = sortedName.name;
        }

        return names;

    }

}

File: 🟢 Easy/2418 - Sort the People/_2418SortThePeople.cpp
class People {
public:
    string name;
    int height;

    People(string name, int height) : name(name), height(height) {}

    static bool heightComparator(People p1, People p2) {
        return p2.height < p1.height;
    }
};

class Solution {

public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        std::vector<People> nameHeight;
        for (int i = 0; i < names.size(); i++) {
            nameHeight.push_back({names[i], heights[i]});
        }

        sort(nameHeight.begin(), nameHeight.end(), People::heightComparator);

        int i = 0;
        for (People sortedName : nameHeight) {
            names[i++] = sortedName.name;
        }

        return names;
    }
};